### DISCUSSION TITLE
Deprecation Notice (reminder): v3 of the artifact actions 📣 (updated)

### DISCUSSION THREAD

**Discussion by samus-aran** [community/community#142581] [Actions]
<img src="https://user-images.githubusercontent.com/19292210/60553865-044dd200-9cea-11e9-859c-d6f266e2f01f.png" width="75%" />

We would like to inform everyone of the **[upcoming deprecation of v3 of the artifact actions](https://github.blog/changelog/2024-04-16-deprecation-notice-v3-of-the-artifact-actions/)**. 📣

This deprecation does not affect GitHub Enterprise Server (GHES).

By **January 30th 2025**, GitHub Actions customers will no longer be able to use v3 of [actions/upload-artifact](https://github.com/actions/upload-artifact) or [actions/download-artifact](https://github.com/actions/download-artifact). To ensure all users have adequate time to transition to the updated version and raise awareness, we will be implementing temporary job failures (brownouts) for any jobs using this deprecated version during scheduled periods. 

* **Deprecated Version**: v3 of [actions/upload-artifact](https://github.com/actions/upload-artifact) and [actions/download-artifact](https://github.com/actions/download-artifact)
* **Brownout Dates**:
    *  January 9th 12pm - 1pm EST
    * January 16th 10am-2pm EST
    * January 23rd 9am-5pm EST

During the brownout periods, any job using this version will **fail temporarily** to signal that the workflow will no longer be supported in the near future. These brownout periods are essential to encourage migration to the supported version and prevent interruptions when the workflow is officially deprecated.

### **What This Means for You**

Please review and update your workflows to use the latest version (v4) of the **artifact actions** as soon as possible. 

Any jobs that we've detected using the deprecated v3 artifact actions will **show an annotation on your workflows**. Below is an example of an indirect dependency showing an annotation at the job level.

![image](https://github.com/user-attachments/assets/ceb0258b-9abb-4d9b-a4f7-a177035cc134)

v4 improves upload and download speeds by up to 98% and includes several new features, however there are [key differences](https://github.com/actions/upload-artifact?tab=readme-ov-file#breaking-changes) from the previous versions that may require updates to your workflows. 

Updated documentation and support resources are available to assist with the migration. 

* [upload-artifact v3 to v4 documentation](https://github.com/actions/upload-artifact/blob/main/docs/MIGRATION.md)
* [download-artifact v3 to v4 documentation](https://github.com/actions/download-artifact/blob/main/docs/MIGRATION.md)

Thank you for your attention! If you have any questions on the migration documentation please comment below.

**Comment by laurent22** [comment]
Why are we told about this just a month before it's taken down? Also GitHub sent us an email telling us we use these actions but we don't, not directly, so it's probably a dependency of a dependency. Since GitHub knows that we use them, why not tell us where exactly so that we can act on it? Right now we have no idea where it's used.

**Reply by samus-aran** [reply]
This deprecation was announced on the [changelog blog back in April](https://github.blog/changelog/2024-04-16-deprecation-notice-v3-of-the-artifact-actions/) and we are setting up these reminders for those that have not migrated from v3 to v4 yet. 

> Also GitHub sent us an email telling us we use these actions but we don't, not directly, so it's probably a dependency of a dependency.

We'll certainly share this with the internal team. As a user of a dependency I would probably want to know that along down the line something will happen but yes it would be helpful to know exactly what would be affected.

**Reply by laurent22** [reply]
As much as I'd like to, I can't subscribe to every blog of every service I use and read every news. If there's an important breaking change coming up, it would be nice to get an email about it in April then, and not just one month before.

**Reply by samus-aran** [reply]
I can understand that!

**Reply by samus-aran** [reply]
@laurent22 could you try out the below code search query to narrow down the jobs within your org? Let us know how you get on. 👍 

`org:[your_org_name] upload-artifact@v3  path:/^\.github\/workflows\//`

or

`org:[your_org_name] download-artifact@v3  path:/^\.github\/workflows\//`

**Reply by laurent22** [reply]
That helped find one workflow, thank you for the tip!

**Comment by ima747** [comment]
How do we identify usage? I found a few references to older versions of our internal actions which in turn consume the upload action, but if we have a nested dependency tree of actions how do we identify what is going to break? Since the email was sent to people who are using outdated versions is there any way to find specifics since something on the GitHub side is seeing our usage of this... as it stands we have to audit everything, and everything that those things consume, on down to the ground and even then not sure how confident we'll be...

**Comment by samus-aran** [comment]
Hi all, we've updated the Brownout Dates on this discussion: 

- January 9th 12pm - 1pm EST
- January 16th 10am-2pm EST
- January 23rd 9am-5pm EST

_[updated] Friday Nov 22_

We also included guidance that you can see annotations at the job level if we've detected there is a use of the v3 artifact actions.

**Reply by ima747** [reply]
Is that guidance generic, and can it be shared here if so? If not, the email I received had those dates but didn't have guidance regarding locating jobs, how could I get that guidance?

**Reply by samus-aran** [reply]
Sorry, I'll clarify that I updated *this* discussion's main copy with this:

> Any jobs that we've detected using the deprecated v3 artifact actions will show an annotation on your workflows. Below is an example of an indirect dependency showing an annotation at the job level.

![image](https://github.com/user-attachments/assets/ceb0258b-9abb-4d9b-a4f7-a177035cc134)

**Comment by cliffchapmanrbx** [comment]
With the [deprecation of the GHES v4 support](https://github.com/github/roadmap/discussions/1014) does this mean GHES has _no planned date_ for this deprecation?

We will of course be moving our GHEC use before the deadline.

**Reply by ChristopherHX** [reply]
> does this mean GHES has _no planned date_ for this deprecation?

In short they can't deprecate it for GHES, it's impossible before v4 exist and the v3 backend remains available as long the changelog says v4 is supported for GHES.
They didn't ship any of their golang based (they leaked this somewhere) backends to GHES yet, they are still migrating github.com over to the new backend at this time old and new backend are both deployed (artifacts v4 as of December 2023) for GitHub.com and logs + step uploaded are sent to both old and new service (as of Jan/Feb 2024).

The services for GitHub Actions on .com are a big mess right now, until 5. December (end of v3 in .com) the date they plan to shutdown their azure pipelines fork (this one is deployed in the GHES VM).

I assume end of december GitHub might make plans for GHES, I guess their timeline might has a regression and it can be > 1a.

BTW I have a custom runner that notices every major protocol change on .com.

Sadly this means the isGHES bomb in artifacts v4 remains :(, even if a fake GHES supports artifacts v4 due to me coding the server part based on the spec and I need to continue to patch it.

**Reply by cliffchapmanrbx** [reply]
Thanks for the context! My team appreciates the work you do to investigate the internal behavior of these systems, it's helped save us time over the years.

**Comment by samus-aran** [comment]
Hi everyone, we've updated the timeline for the deprecation of v3 of artifact actions. Please see the top post for these updates including scheduled Brownout periods. 👀

**Comment by ChuchinCorreao** [comment]
> <img alt="" width="75%" src="https://user-images.githubusercontent.com/19292210/60553865-044dd200-9cea-11e9-859c-d6f266e2f01f.png">
> 
> We would like to inform everyone of the **[upcoming deprecation of v3 of the artifact actions](https://github.blog/changelog/2024-04-16-deprecation-notice-v3-of-the-artifact-actions/)**. 📣
> 
> This deprecation does not affect GitHub Enterprise Server (GHES).
> 
> By **January 30th 2025**, GitHub Actions customers will no longer be able to use v3 of [actions/upload-artifact](https://github.com/actions/upload-artifact) or [actions/download-artifact](https://github.com/actions/download-artifact). To ensure all users have adequate time to transition to the updated version and raise awareness, we will be implementing temporary job failures (brownouts) for any jobs using this deprecated version during scheduled periods.
> 
> * **Deprecated Version**: v3 of [actions/upload-artifact](https://github.com/actions/upload-artifact) and [actions/download-artifact](https://github.com/actions/download-artifact)
> * **Brownout Dates**:
>   
>   * January 9th 12pm - 1pm EST
>   * January 16th 10am-2pm EST
>   * January 23rd 9am-5pm EST
> 
> During the brownout periods, any job using this version will **fail temporarily** to signal that the workflow will no longer be supported in the near future. These brownout periods are essential to encourage migration to the supported version and prevent interruptions when the workflow is officially deprecated.
> 
> ### **What This Means for You**
> Please review and update your workflows to use the latest version (v4) of the **artifact actions** as soon as possible.
> 
> Any jobs that we've detected using the deprecated v3 artifact actions will **show an annotation on your workflows**. Below is an example of an indirect dependency showing an annotation at the job level.
> 
> ![image](https://private-user-images.githubusercontent.com/4041275/383659612-ceb0258b-9abb-4d9b-a4f7-a177035cc134.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ3ODMyOTMsIm5iZiI6MTczNDc4Mjk5MywicGF0aCI6Ii80MDQxMjc1LzM4MzY1OTYxMi1jZWIwMjU4Yi05YWJiLTRkOWItYTRmNy1hMTc3MDM1Y2MxMzQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIyMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMjFUMTIwOTUzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MzM4MmY0YzcxNTY1Y2QxNmM4ZDcwMzMxNTliOGYxODZhMzE3NjM4MDlmMWU5MTQ1NjAwZWVmN2Y2NGZhYjAwOCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.XbXy3DBn5iFM82pkdpDthudNe4qdtQH95xVtjwnHm40)
> 
> v4 improves upload and download speeds by up to 98% and includes several new features, however there are [key differences](https://github.com/actions/upload-artifact?tab=readme-ov-file#breaking-changes) from the previous versions that may require updates to your workflows.
> 
> Updated documentation and support resources are available to assist with the migration.
> 
> * [upload-artifact v3 to v4 documentation](https://github.com/actions/upload-artifact/blob/main/docs/MIGRATION.md)
> * [download-artifact v3 to v4 documentation](https://github.com/actions/download-artifact/blob/main/docs/MIGRATION.md)
> 
> Thank you for your attention! If you have any questions on the migration documentation please comment below.

/

**Comment by ChuchinCorreao** [comment]
Uyg

**Comment by msluther** [comment]
How do we get access to the deprecation annotation programatically. I can see them if I happen to go look at my GHAction output, but I'm trying to build a job that runs periodically and scans/notifies for these sorts of things, but they appear to not come back with any of the annotations associated with that check-suite (e.g. https://api.github.com/repos/ORG/REPO/check-suites/CHECK-SUITE-ID/check-runs). The deprecation annotation here seems to be something special and accumulating data across different check-runs.  What's the github api endpoint necessary to find these custom annotations?

**Reply by yacaovsnc** [reply]
I am able to query this [API](https://docs.github.com/en/rest/checks/runs?apiVersion=2022-11-28#list-check-run-annotations) to get the annotations, they are on the check-run.

**Reply by msluther** [reply]
In your case do you have one check-run within the suite that has the deprecation or multiple? The repo I was looking at had multiple and the annotation that rendered in the web UI for the check suite had some synthetic annotation that accumulated data from multiple check runs. e.g. the web UI shows me:

```
[Deprecation notice: v1, v2, and v3 of the artifact actions](https://github.com/ORG/REPO/actions/runs/RUN_NUMBER/workflow)
The following artifacts were uploaded using a version of actions/upload-artifact that is scheduled for deprecation: "ARTIFACT_1", "ARTIFACT_2".
Please update your workflow to use v4 of the artifact actions.
Learn more: https://github.blog/changelog/2024-04-16-deprecation-notice-v3-of-the-artifact-actions/
```

Those two artifacts were from different jobs within the workflow. No other annotation we have does any sort of accumulation like this and _those_ I can get just fine. Not that the deprecation link goes to my workflow run itself and NOT the particular check-run

**Reply by yacaovsnc** [reply]
I see, I am looking at a different annotation.

```
[
	{
		"path": ".github",
		"blob_href": ...,
		"start_line": 48,
		"start_column": null,
		"end_line": 48,
		"end_column": null,
		"annotation_level": "failure",
		"title": "",
		"message": "This request has been automatically failed because it uses a deprecated version of `actions/upload-artifact: v3`. Learn more: https://github.blog/changelog/2024-04-16-deprecation-notice-v3-of-the-artifact-actions/",
		"raw_details": ""
	}
]
```

This one above is associated with check-run.  The one you are checking is associated with a suite and is not returned.

**Reply by msluther** [reply]
Right, mine was the warning and not a failiure during the brownout. I don't see any github APIs that return annotations for the suite only checkruns. Are the docs wrong? How do I get the "suite-level" annotations so that I can see these sorts of warnings that I can see in the web ui?

**Reply by yacaovsnc** [reply]
Unfortunately I don't think there is an API to expose check-suite level annotations at this point.  😞

**Reply by msluther** [reply]
That is... unfortunate and means everyone has to go look at their github workflow runs by hand to catch this and they likely won't do that and wont' see it until they get hit with one of the brownouts (which is when we noticed). 😭

**Comment by joey-cyt** [comment]
I have a workflow with the following:
```
    - name: Upload deb package
       uses: actions/upload-artifact@v4
       with:
         name: myApp-${{ env.VERSION }}-ub2204.deb
         path: ./*deb
         compression-level: 0
 
     - name: Upload tgz package
       uses: actions/upload-artifact@v4
       with:
         name: myApp-${{ env.VERSION }}-ub2204.tgz
         path: ./*tgz
         compression-level: 0
```

It built successfully in this early morning. However it started failing since {brownout time}. 
I'm confused about the deprecated v3 error because the workflow file uses v4 upload-artifact. Are there something should do for this migration? (I deleted actions->cache, no help).
The following is the build log with debug info:
```
Download action repository 'actions/upload-artifact@v4' (SHA:65c4c4a1ddee5b72f698fdd19549f0f0fb45cf08)
##[debug]Download 'https://api.github.com/repos/actions/upload-artifact/tarball/65c4c4a1ddee5b72f698fdd19549f0f0fb45cf08' to '/home/runner/work/_actions/_temp_6a49f1da-15d1-4a72-a7aa-8bd9f94b5624/d12bebf3-bc32-472c-8b34-b777301deaa8.tar.gz'
##[debug]Unwrap 'actions-upload-artifact-65c4c4a' to '/home/runner/work/_actions/actions/upload-artifact/v4'
##[debug]Archive '/home/runner/work/_actions/_temp_6a49f1da-15d1-4a72-a7aa-8bd9f94b5624/d12bebf3-bc32-472c-8b34-b777301deaa8.tar.gz' has been unzipped into '/home/runner/work/_actions/actions/upload-artifact/v4'.
##[debug]Action 'actions/upload-artifact@v4' already downloaded at '/home/runner/work/_actions/actions/upload-artifact/v4'.
##[debug]action.yml for action: '/home/runner/work/_actions/actions/checkout/v4/action.yml'.
##[debug]action.yml for action: '/home/runner/work/_actions/awalsh128/cache-apt-pkgs-action/latest/action.yml'.
Getting action download info
Error: This request has been automatically failed because it uses a deprecated version of `actions/upload-artifact: v3`. Learn more: https://github.blog/changelog/2024-04-16-deprecation-notice-v3-of-the-artifact-actions/
```

**Reply by yacaovsnc** [reply]
Check your third party dependencies, the `latest` tag on https://github.com/awalsh128/cache-apt-pkgs-action/blob/latest/action.yml still uses `v3`.

**Reply by joey-cyt** [reply]
Changing the version of cache-apt-pkgs-action from fake "latest" to true latest v1.4.3 fixed the issue.
Thank you very much @yacaovsnc

**Comment by MSRS-Shop** [comment]
Their update to v4 seems to have broke using ansible get_url authorization header format. I've had to change my scripts to use ansible curl instead. Any one else having this issue?

**Reply by yacaovsnc** [reply]
Are you using APIs to upload/download artifacts?  v3 and v4 have different backends so they authenticate differently.

**Reply by MSRS-Shop** [reply]
WORKS:

> - name: Get artifacts for the workflow run
  uri:
    url: "https://api.github.com/repos/{{ repo_owner }}/{{ repo_name }}/actions/runs/{{ workflow_run_id }}/artifacts"
    method: GET
    headers:
      Authorization: "Bearer {{ github_token }}"
      Accept: "application/vnd.github+json"
    return_content: yes
    status_code: 200
  register: artifacts_info

DOES NOT WORK
> - name: Download the artifact from the workflow run
  get_url:
    url: "{{ artifact_url_application }}"
    dest: "{{ download_dest_application }}/{{ artifact_name_application }}.zip"
    mode: '0755'
    owner: pi
    group: pi
    headers:
      Accept: "application/vnd.github+json"
      Authorization: "Bearer {{ github_token }}"
      X-GitHub-Api-Version: "2022-11-28"
    force_basic_auth: yes
    use_netrc: false
  register: download_result

The only difference is the get_url method. I added API version, force_basic_auth, and use_netrc during troubleshooting

**Reply by yacaovsnc** [reply]
For v4, the download url you received from the GET call has a shared access signature, and you should not add any additional auth headers when downloading.

**Comment by rzr** [comment]
Just thinking loud for anyone cares, can I suggest an option in project configuration to allow overloading action files with a pattern ?
I could sounds hackish but that would be a fast way to allow (re)-building earlier revision of project without patching it.

**Comment by wqw547243068** [comment]
My github page website suddenly failed to build for the sake of this update (Deprecation Notice of v3)

[Error message](https://github.com/wqw547243068/wqw547243068.github.io/actions/runs/13215389884/job/36894042407):
- [Set up job](https://github.com/wqw547243068/wqw547243068.github.io/actions/runs/13215389884/job/36894042407#annotation:1:30)
This request has been automatically failed because it uses a deprecated version of `actions/upload-artifact: v3`. Learn more: https://github.blog/changelog/2024-04-16-deprecation-notice-v3-of-the-artifact-actions/

However, I didn't find any clu about `actions/upload-artifact: v3` in configuration file [.github/workflow/pages.yml](https://github.com/wqw547243068/wqw547243068.github.io/blob/master/.github/workflows/pages.yml)

Key information:

```yml
jobs:
  # Build job
  build:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout
        uses: actions/checkout@v3
      - name: Setup Pages
        uses: actions/configure-pages@v2
      - name: Build with Jekyll
        uses: actions/jekyll-build-pages@v1
        with:
          source: ./
          destination: ./_site
      - name: Upload artifact
        uses: actions/upload-pages-artifact@v1
```

How should I update this configuration file ?

Thanks a lot !

**Reply by ChristopherHX** [reply]
Bump the `@v<no>` of the pages actions for example sync your uses statements to look like mine.

(Setup dependabot for actions and upgrade earlier by automatically created PRs)
```yaml
      - name: Setup Pages
        if: github.event_name == 'push'
        uses: actions/configure-pages@v5
      - name: Upload artifact
        if: github.event_name == 'push'
        uses: actions/upload-pages-artifact@v3
        with:
          path: 'webapp'
      - name: Deploy to GitHub Pages
        if: github.event_name == 'push'
        id: deployment
        uses: actions/deploy-pages@v4
```

Make sure to update deploy-pages as well as this might be required to use a similar artifacts backend

_The error message is garbage you are totally correct about this_

**Reply by wqw547243068** [reply]
> Bump the `@v<no>` of the pages actions for example sync your uses statements to look like mine.例如，将页面作的 `@v<no>` 同步，使其看起来像我的。
> 
> (Setup dependabot for actions and upgrade earlier by automatically created PRs)（通过自动创建的 PR 为 actions 设置 dependabot 并提前升级）
> 
> ```yaml
>       - name: Setup Pages
>         if: github.event_name == 'push'
>         uses: actions/configure-pages@v5
>       - name: Upload artifact
>         if: github.event_name == 'push'
>         uses: actions/upload-pages-artifact@v3
>         with:
>           path: 'webapp'
>       - name: Deploy to GitHub Pages
>         if: github.event_name == 'push'
>         id: deployment
>         uses: actions/deploy-pages@v4
> ```
> 
> Make sure to update deploy-pages as well as this might be required to use a similar artifacts backend确保更新 deploy-pages，这可能是使用类似工件后端所必需的
> 
> _The error message is garbage you are totally correct about this错误消息是垃圾，您对此完全正确_

I've updated configuration as following

```yaml
jobs:
  # Build job
  build:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout
        uses: actions/checkout@v3
      - name: Setup Pages
        # uses: actions/configure-pages@v2
        if: github.event_name == 'push'
        uses: actions/configure-pages@v5
      - name: Build with Jekyll
        uses: actions/jekyll-build-pages@v1
        with:
          source: ./
          destination: ./_site
      - name: Upload artifact
        #uses: actions/upload-pages-artifact@v1
        if: github.event_name == 'push'
        uses: actions/upload-pages-artifact@v3
```

And it works !  Thank you very much !

**Comment by hlwndon** [comment]
> <img alt="" width="75%" src="https://user-images.githubusercontent.com/19292210/60553865-044dd200-9cea-11e9-859c-d6f266e2f01f.png">
> 
> We would like to inform everyone of the **[upcoming deprecation of v3 of the artifact actions](https://github.blog/changelog/2024-04-16-deprecation-notice-v3-of-the-artifact-actions/)**. 📣
> 
> This deprecation does not affect GitHub Enterprise Server (GHES).
> 
> By **January 30th 2025**, GitHub Actions customers will no longer be able to use v3 of [actions/upload-artifact](https://github.com/actions/upload-artifact) or [actions/download-artifact](https://github.com/actions/download-artifact). To ensure all users have adequate time to transition to the updated version and raise awareness, we will be implementing temporary job failures (brownouts) for any jobs using this deprecated version during scheduled periods.
> 
> * **Deprecated Version**: v3 of [actions/upload-artifact](https://github.com/actions/upload-artifact) and [actions/download-artifact](https://github.com/actions/download-artifact)
> * **Brownout Dates**:
>   
>   * January 9th 12pm - 1pm EST
>   * January 16th 10am-2pm EST
>   * January 23rd 9am-5pm EST
> 
> During the brownout periods, any job using this version will **fail temporarily** to signal that the workflow will no longer be supported in the near future. These brownout periods are essential to encourage migration to the supported version and prevent interruptions when the workflow is officially deprecated.
> 
> ### **What This Means for You**
> Please review and update your workflows to use the latest version (v4) of the **artifact actions** as soon as possible.
> 
> Any jobs that we've detected using the deprecated v3 artifact actions will **show an annotation on your workflows**. Below is an example of an indirect dependency showing an annotation at the job level.
> 
> ![image](https://private-user-images.githubusercontent.com/4041275/383659612-ceb0258b-9abb-4d9b-a4f7-a177035cc134.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzkxOTQ2MjksIm5iZiI6MTczOTE5NDMyOSwicGF0aCI6Ii80MDQxMjc1LzM4MzY1OTYxMi1jZWIwMjU4Yi05YWJiLTRkOWItYTRmNy1hMTc3MDM1Y2MxMzQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDIxMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAyMTBUMTMzMjA5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NGE3Y2ViNzU2Yjc1NWEyYmUzZjEzYTYwMGY4MDMzMDU3YjIwYjE3MmUwZmU2Y2E1ZTc2YjQ1Y2Y0ODY1NWZmZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.KuaXbrL3zDYmHrimLwx3_7QM_n9U7moqQP8s-DCmEq4)
> 
> v4 improves upload and download speeds by up to 98% and includes several new features, however there are [key differences](https://github.com/actions/upload-artifact?tab=readme-ov-file#breaking-changes) from the previous versions that may require updates to your workflows.
> 
> Updated documentation and support resources are available to assist with the migration.
> 
> * [upload-artifact v3 to v4 documentation](https://github.com/actions/upload-artifact/blob/main/docs/MIGRATION.md)
> * [download-artifact v3 to v4 documentation](https://github.com/actions/download-artifact/blob/main/docs/MIGRATION.md)
> 
> Thank you for your attention! If you have any questions on the migration documentation please comment below.

**Comment by hlwndon** [comment]
01322406241

**Comment by TorokEni** [comment]
Hi everyone,
I would really like to update to v4 but we are using the enterprise edition of github, any news about when it will be available also for GHE?
Thank you!
